packages/opal-client/opal_client/data/updater.py

```diff
@@ -296,7 +297,7 @@ def calc_hash(data):
         """
         try:
             if not isinstance(data, str):
-                data = json.dumps(data, default=str)
+                data = json.dumps(data, default=pydantic_encoder)
```

Comment:
Can you please explain how does that change the json serialization behavior?

Reply:
used `jsonable_encoder` instead of `pydantic_encoder` and also created a util method for the serialization with exclude_none=True